pkg/dnshelpers/util.go

```diff
@@ -70,15 +70,28 @@ func GetPreferredInternalIPAddressForNodeName(network *configv1.Network, node *c
 	return "", "", fmt.Errorf("no matches found for ip family %q for node %q", ipFamily, node.Name)
 }
 
+// GetPreferredIPFamily checks network status for service CIDR to conclude IP family. If status is not yet populated
+// fallback to spec.
 func GetPreferredIPFamily(network *configv1.Network) (string, error) {
-	if len(network.Status.ServiceNetwork) == 0 || len(network.Status.ServiceNetwork[0]) == 0 {
-		return "", fmt.Errorf("networks.%s/cluster: status.serviceNetwork not found", configv1.GroupName)
+	var serviceCIDR string
+	switch {
+	case len(network.Status.ServiceNetwork) != 0 && len(network.Status.ServiceNetwork[0]) != 0:
+		serviceCIDR = network.Status.ServiceNetwork[0]
+		if len(serviceCIDR) == 0 {
+			return "", fmt.Errorf("networks.%s/cluster: status.serviceNetwork[0] is empty", configv1.GroupName)
+		}
```

Comment:
I know this is a cherry-pick, but L81 will never be reached, because of the condition for the case on L78 already eliminates if the length of [0] is zero.